src/modules/adaptors/Morpho/MorphoBlue/MorphoBlueSupplyAdaptor.sol

```diff
@@ -0,0 +1,234 @@
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("Morpho Blue Supply Adaptor V 0.1"));
+    }
+
+    //============================================ Implement Base Functions ===========================================
+
+    /**
+     * @notice Allows user to deposit into MB markets, only if Cellar has a MBSupplyAdaptorPosition as its holding position.
+     * @dev Cellar must approve Morpho Blue to spend its assets, then call deposit to lend its assets.
+     * @param assets the amount of assets to lend on Morpho Blue.
+     * @param adaptorData adaptor data containing the abi encoded Morpho Blue market.
+     * @dev configurationData is NOT used.
+     */
+    function deposit(uint256 assets, bytes memory adaptorData, bytes memory) public override {
+        MarketParams memory market = abi.decode(adaptorData, (MarketParams));
+        ERC20 loanToken = ERC20(market.loanToken);
+        loanToken.safeApprove(address(morphoBlue), assets);
+        _deposit(market, assets, address(this));
+
+        // Zero out approvals if necessary.
+        _revokeExternalApproval(loanToken, address(morphoBlue));
+    }
+
+    /**
+     * @notice Cellars must withdraw from Morpho Blue lending market, then transfer assets to receiver.
+     * @dev Important to verify that external receivers are allowed if receiver is not Cellar address.
+     * @param assets the amount of assets to withdraw from Morpho Blue lending market
+     * @param receiver the address to send withdrawn assets to
+     * @param adaptorData adaptor data containing the abi encoded Morpho Blue market.
+     * @dev configurationData is NOT used
+     */
+    function withdraw(uint256 assets, address receiver, bytes memory adaptorData, bytes memory) public override {
+        // Run external receiver check.
+        _externalReceiverCheck(receiver);
+        MarketParams memory market = abi.decode(adaptorData, (MarketParams));
+        // Withdraw assets from Morpho Blue.
+        _validateMBMarket(market);
+        _withdraw(market, assets, receiver);
+    }
+
+    /**
+     * @notice Returns the amount of loanToken that can be withdrawn.
+     * @dev Compares loanToken supplied to loanToken borrowed to check for liquidity.
+     *      - If loanToken balance is greater than liquidity available, it returns the amount available.
+     * @param adaptorData adaptor data containing the abi encoded Morpho Blue market.
+     * @return withdrawableSupply liquid amount of `loanToken` cellar has lent to specified MB market.
+     */
+    function withdrawableFrom(
+        bytes memory adaptorData,
+        bytes memory
+    ) public view override returns (uint256 withdrawableSupply) {
+        MarketParams memory market = abi.decode(adaptorData, (MarketParams));
+        (uint256 totalSupplyAssets, , uint256 totalBorrowAssets, ) = morphoBlue.expectedMarketBalances(market);
+        if (totalBorrowAssets == totalSupplyAssets) return 0;
```

Comment:
Is it ever possible for totalBorrowAssets to be greater than totalSupplyAssets? If so this should probably be `if (totalBorrowAssets >= totalSupplyAssets)`

Reply:
The Morpho team actually pointed out to make this `==` vs `>=` since `totalBorrowAssets` cannot be greater than `totalSupplyAssets` as per [here](https://github.com/PeggyJV/cellar-contracts/pull/154#discussion_r1445835934).

Reply:
I think this is much safer, yes it is extremely unlikely for their to be a bug that would cause `totalBorrowAssets` to be greater than `totalSupplyAssets`, but if it ever does happen, leaving it as `==` means user withdraws will revert if we try to pull from the morpho blue supply position. The strategist would be able to recover the cellar from this state, but considering that making it `>=` would only add like 3 gas to the call, I think it is much better to go with the more robust design.